Common/Product/ReplWindow/Repl/ReplWindow.cs

```diff
@@ -2110,14 +2110,18 @@ private sealed class CommandFilter : IOleCommandTarget {
         /// <summary>
         /// Appends text to the output buffer and updates projection buffer to include it.
         /// </summary>
-        internal void AppendOutput(ConsoleColor color, string text, bool lastOutput) {
+        internal void AppendOutput(OutputBuffer.OutputEntry[] entries) {
+            if (!entries.Any())
+                return;
+
+            var spans = new List<KeyValuePair<Span, OutputBuffer.OutputEntryProperties>>(entries.Length);
```

Comment:
This has to be outside of the using block here but I'll move it down a few lines.